src/tools/common/sss_tools.c

```diff
@@ -333,13 +333,15 @@ errno_t sss_tool_route(int argc, const char **argv,
                 return tool_ctx->init_err;
             }
 
-            ret = tool_cmd_init(tool_ctx, &commands[i]);
-            if (ret != EOK) {
-                DEBUG(SSSDBG_FATAL_FAILURE,
-                      "Command initialization failed [%d] %s\n",
-                      ret, sss_strerror(ret));
-                return ret;
-            }
+	    if(strcmp(argv[2],"--help")){
```

Comment:
This line looks like a bug at first glance. Please use ``if (strcmp(argv[2],"--help") != 0) {`` to make the intention clear.